### ISSUE TITLE
Licensing issue: files jni_md.h

### ISSUE THREAD

**Issue by pgrt** [xerial/sqlite-jdbc#705]
Hello,

I am a Debian developer, and I would like to package sqlite-jdbc in Debian. The library would be a great addition to Debian as we see many other software depending on it, and that we cannot package at the moment.

Yet there is a licensing issue in the lib/inc_linux/jni_md.h file (and likewise for inc_win and inc_mac):

 * Copyright (c) 2006, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.

Do you think it might be possible to get rid of the second line and having the file licensed under a free software license? That could unblock the licensing issue from our side and I believe it would be better for all users, even outside Debian or Linux.

Thanks a lot for your attention,

Best regards,

Pierre

**Comment by gotson** [comment]
> Do you think it might be possible to get rid of the second line and having the file licensed under a free software license?

One cannot change a license just because they want to. This file is part of the JDK. However the license seemed to have been garbled, it should be available under GPL v2, at least the version I found in openjdk 8 seemed to be. 

Note also that this is not part of the distributable t. Not sure what you intended to package for Debian but AFAIK this is used to build the native libs only. It shouldn't be needed as part of the jar, since those files are part of the JDK that will run the jar.

**Comment by pgrt** [comment]
Hi,

Thanks for answering me so quickly!

Ok, I get it, this file comes from an ancient version of the JDK... so of course you cannot change it. Like you, I checked the license of this file in my openjdk and it is also GPL-2.

I should have given you more information about my goals. i would like to be able to design a Debian source package that builds sqlite-jdbc from source to provide both the jni and the java part as Debian packages. We need to build from source to show our users they really run free software on their computers (we show the sources!). And this is also why we need to have every piece of source code be free.

Certainly I can thus drop the lib/inc_linux/jni_md.h file and include the jni_md.h file from openjdk instead, in order to skirt this licensing issue.

Thanks again for the explanations,

Pierre

**Comment by gotson** [comment]
If you look at the makefile common, it is used only for building the native lib for other arch, like building a windows lib from a Mac, in which case you need the md file from the windows jdk.

If you only build native, I think it should use the one from the JDK you have, via JAVA_HOME.

Xerial is away for now, but I think we should be able to replace the md files with some that have the proper license. Anyway the whole jdk was licensed as GPL v2 with classpath exception, so this file should also be like that.

**Comment by pgrt** [comment]
Hello again,

Thanks for the information. Thus I should use the JDK I have, this is ok regarding my need.

I am closing the issue, thanks again for the discussion.

Best,
Pierre
